# Generated by Django 4.2.2 on 2023-06-14 10:59

from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('users', '0003_remove_follow_unique_users_follow_unique_users_and_more'),
    ]

    operations = [
        migrations.RemoveConstraint(
            model_name='follow',
            name='unique_users',
        ),
        migrations.RemoveConstraint(
            model_name='follow',
            name='different_users',
        ),
        migrations.AddConstraint(
            model_name='follow',
            constraint=models.UniqueConstraint(fields=('user', 'author'), name='unique_users', violation_error_message='Такая комбинация пользователей уже существует.'),
        ),
        migrations.AddConstraint(
            model_name='follow',
            constraint=models.CheckConstraint(check=models.Q(('user', models.F('author')), _negated=True), name='different_users', violation_error_message='Автор и подписчик должны быть разными людьми.'),
        ),
    ]
